### PR TITLE
fix(ziti): handle managed identities

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- parse managed agent identity names when ziti management returns NotFound
- prefer dialer identity IDs over source identifiers for ziti connections
- add unit coverage for managed identity parsing and ziti connection lookup

## Testing
- buf generate buf.build/agynio/api --include-imports --path agynio/api/agents/v1 --path agynio/api/apps/v1 --path agynio/api/threads/v1 --path agynio/api/chat/v1 --path agynio/api/notifications/v1 --path agynio/api/files/v1 --path agynio/api/agent_state/v1 --path agynio/api/token_counting/v1 --path agynio/api/llm/v1 --path agynio/api/secrets/v1 --path agynio/api/identity/v1 --path agynio/api/tracing/v1 --path agynio/api/users/v1 --path agynio/api/organizations/v1 --path agynio/api/ziti_management/v1 --path agynio/api/gateway/v1
- go vet ./...
- go test ./...

#43

### CODE REVIEW

File: internal/identity/managed_identity.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	trimmed := strings.TrimSpace(name)
+	if trimmed == "" {
+		return ResolvedIdentity{}, false
+	}
+	if !strings.HasPrefix(trimmed, managedAgentPrefix) {
+		return ResolvedIdentity{}, false
+	}
+
+	remainder := strings.TrimPrefix(trimmed, managedAgentPrefix)
+	separator := strings.LastIndex(remainder, "-")
```
Comment (noa-lucent) [inline internal/identity/managed_identity.go]: [major] `strings.LastIndex(remainder, \"-\")` is the wrong tool for locating the UUID boundary. It works only when the suffix after the UUID is a single non-hyphenated segment (like `acde1234`). If the Ziti suffix ever contains a hyphen (e.g. `agent-{UUID}-ab-cd`), `LastIndex` will split inside the suffix, `uuid.Parse` will reject the mangled candidate, and a valid managed identity will silently fail to resolve.

UUIDs have a fixed length of 36 characters. Parsing by length is deterministic and immune to suffix format:

```go
const uuidLen = 36

remainder := strings.TrimPrefix(trimmed, managedAgentPrefix)
if len(remainder) < uuidLen {
    return ResolvedIdentity{}, false
}

agentID := remainder[:uuidLen]
if _, err := uuid.Parse(agentID); err != nil {
    return ResolvedIdentity{}, false
}
```

This also removes the need for the `separator <= 0` guard.

File: internal/identity/managed_identity_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	if _, ok := ParseManagedIdentityName(""); ok {
+		t.Fatalf("expected empty identity to be rejected")
+	}
+	if _, ok := ParseManagedIdentityName("svc-runner-acde1234"); ok {
+		t.Fatalf("expected non-agent identity to be rejected")
+	}
+	if _, ok := ParseManagedIdentityName("agent-not-a-uuid-acde1234"); ok {
+		t.Fatalf("expected invalid uuid to be rejected")
+	}
+}
```
Comment (noa-lucent) [inline internal/identity/managed_identity_test.go]: [minor] A few edge cases worth adding to lock in behavior:

1. **No suffix — bare `agent-{UUID}`**: Currently returns `false` due to the `LastIndex` split landing inside the UUID. Whether this should succeed or fail is a design choice, but either way it should be documented with a test.

2. **Hyphenated suffix — `agent-{UUID}-ab-cd`**: This is the case that breaks the current `LastIndex` approach (see other comment). If you switch to length-based parsing, add a test confirming it succeeds.

3. **Prefix-only — `agent-`**: Exercises the `separator <= 0` / short-remainder guard.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The `ziticonn` changes (dialer identity preference + tests) carry over cleanly from #122 and look good.

The new `managed_identity.go` name-parsing fallback is a solid idea for the NotFound recovery path in `client.go`, and the gRPC `codes.NotFound` check is appropriately narrow.

**Requesting changes** for one structural issue: the `strings.LastIndex`-based UUID extraction is fragile and will silently fail for any hyphenated Ziti suffix. Since UUIDs are fixed-length (36 chars), parsing by length is deterministic and robust. See inline comment for the suggested fix.

Also two minor suggestions for additional test coverage.

**Review by noa-lucent — APPROVED** [review]
All three prior comments addressed:

1. **`managed_identity.go`** — `LastIndex` replaced with fixed-length UUID extraction (`managedAgentUUIDSize = 36`). The `+2` minimum-length check neatly enforces both the separator dash and a non-empty suffix. Clean and deterministic.

2. **`managed_identity_test.go`** — Refactored to table-driven tests with all requested edge cases: hyphenated suffix, no suffix, prefix-only, empty suffix, plus whitespace trimming. Thorough coverage.

3. **`conn_test.go`** — `dialerOnlyConn` type and `TestSourceIdentityFromConnDialerOnlyEmptyFallsThrough` added, exercising the dialer-only fallthrough path.

LGTM.
